src/node_type.rs

```diff
@@ -50,37 +49,37 @@ pub struct NodeKey {
 
 impl NodeKey {
     /// Creates a new `NodeKey`.
-    pub fn new(version: Version, nibble_path: NibblePath) -> Self {
+    pub(crate) fn new(version: Version, nibble_path: NibblePath) -> Self {
```

Comment:
Since the NodeKey is public (so that it can be written to the backing store), I made all the methods except `encode` and `decode` be `pub(crate)`, so they don't appear in the public API.